ColorPatternsTests/Scenes/Game/GameViewModelTests.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  GameViewModelTests.swift
+//  ColorPatternsTests
+//
+//  Created by Marek Skrzelowski on 22/01/2020.
+//  Copyright © 2020 vandermesis. All rights reserved.
+//
+
+import Quick
+import Nimble
+
+@testable import ColorPatterns
+
+final class GameViewModelTests: QuickSpec {
+
+    override func spec() {
+
+        var gameEngine: GameEngine!
+        var userScore: Score!
+        var viewModel: GameViewModel!
+
+        beforeEach {
+            gameEngine = GameEngine()
+            userScore = Score()
+            viewModel = GameViewModel(gameEngine: gameEngine, score: userScore)
+        }
+
+        describe("pressing color bar") {
+
+            beforeEach {
+                gameEngine.colorsArray = [0.1, 0.2, 0.3, 0.4, 0.5]
+                viewModel.didPressColorBar(colorBar: 1)
+            }
+
+            it("should call game engine save picked color") {
+                expect(gameEngine.pickedColor).to(equal(gameEngine.colorsArray[1]))
+            }
+
+            it("should call userScore to get score 0") {
+                expect(userScore.score).to(equal(0))
+            }
+
+            it("should call game engine to save user color") {
+                expect(gameEngine.userColor).to(equal(gameEngine.pickedColor))
+            }
+
+            // FIXME: That test doesn't work if GameEngine and Score are structs,
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/vandermesis/ColorPatterns/blob/4955f1fb2a87cdbcba8f9e189d9a68e1bb2f4ac3/ColorPatternsTests/Scenes/Game/GameViewModelTests.swift#L47"></span>FIXMEs should be resolved (That test doesn’t work if Game…).<br />
<code>todo</code> <code>GameViewModelTests.swift:47</code></td>
    </tr>
  </tbody>
</table>